src/utils/updateConfigurations.js

```diff
@@ -108,6 +108,96 @@ export const createNewDenominator = (configurations, newDenominatorInfo) => {
     return configurationsToSave
 }
 
+// create new denominator relation
+export const updateDenominatorRelations = ({
+    configurations,
+    relation,
+    updateType,
+    newCode,
+}) => {
+
+    const metaDataVersion = configurations.metaDataVersion
+    const numerators = configurations.numerators
+    const coreIndicators = configurations.coreIndicators
+    const dataSets = configurations.dataSets
+    const denominatorRelations = configurations.denominatorRelations
+    const denominators = configurations.denominators
+    const externalRelations = configurations.externalRelations
+    const numeratorRelations = configurations.numeratorRelations
+    const groups = configurations.groups
+    
+    // construct a numerator object
+    const newDenominatorRelation = {
+        A: relation.A,
+        B: relation.B,
+        code: newCode,
+        name: relation.name,
+        criteria: relation.criteria,
+        type: relation.type,
+    }
+
+
+
+    let configurationsToSave = {}
+
+    switch (updateType) {
+        case 'update':
+            configurationsToSave = {
+                metaDataVersion,
```

Comment:
you don't need to repeat all of these individual items, you can just do 
`configurationsToSave = {...configurations, denominatorRelations: updateDenominatorRelation(denominatorRelations,relation)}`

Using the spread operator will copy all the properties, and then you can just overwrite denominatorRelations by putting it after that spread operation.